src/analog/m2kanalogin_impl.cpp

```diff
@@ -37,7 +37,8 @@ using namespace std::placeholders;
 M2kAnalogInImpl::M2kAnalogInImpl(iio_context * ctx, std::string adc_dev, bool sync, M2kHardwareTrigger *trigger) :
 	M2kAnalogIn(),
 	m_need_processing(false),
-	m_trigger(trigger)
+	m_trigger(trigger),
+	m_max_samplerate(-1)
```

Comment:
These initializations should be switched one with the other, in order to suppress the order warning.